### PR TITLE
feat(core): reject samples with Nano/Micro second timestamps

### PR DESCRIPTION
**Pull request by TanviBhavsar**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [x] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

Reject samples before 1 month and after 1 day from current time

### CODE REVIEW

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 1]
```diff
@@ -119,10 +121,19 @@ extends ChunkMap(memFactory, initMapSize) with ReadablePartition {
   def ingest(row: RowReader, blockHolder: BlockMemFactory): Unit = {
     // NOTE: lastTime is not persisted for recovery.  Thus the first sample after recovery might still be out of order.
     val ts = dataset.timestamp(row)
+    val currentEpochTime : Long = Instant.now.getEpochSecond()
+    // Reject samples before 1 month and after 1 day from current time
+    val lowerLimit = currentEpochTime - (31 * 24 * 60 * 60)
+    val upperLimit = currentEpochTime + (24 * 60 * 60 )
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: Need to make the 1 month, and 1 day configurable; and dont calculate each time.

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 2]
```diff
@@ -119,10 +121,19 @@ extends ChunkMap(memFactory, initMapSize) with ReadablePartition {
...
+
     if (ts < timestampOfLatestSample) {
       shardStats.outOfOrderDropped.increment
       return
     }
+     else if (((ts/1000) < lowerLimit) || ((ts/1000) > upperLimit)) {
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: incoming timestamp is already in millisecond. why divide by 1000 ?

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 3]
```diff
@@ -119,10 +121,19 @@ extends ChunkMap(memFactory, initMapSize) with ReadablePartition {
   def ingest(row: RowReader, blockHolder: BlockMemFactory): Unit = {
     // NOTE: lastTime is not persisted for recovery.  Thus the first sample after recovery might still be out of order.
     val ts = dataset.timestamp(row)
+    val currentEpochTime : Long = Instant.now.getEpochSecond()
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: It should be in millis. System.currentTimeMillis() ?

File: core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala [hunk 4]
```diff
@@ -119,10 +121,19 @@ extends ChunkMap(memFactory, initMapSize) with ReadablePartition {
...
+    // Reject samples before 1 month and after 1 day from current time
+    val lowerLimit = currentEpochTime - (31 * 24 * 60 * 60)
+    val upperLimit = currentEpochTime + (24 * 60 * 60 )
+
     if (ts < timestampOfLatestSample) {
       shardStats.outOfOrderDropped.increment
       return
     }
+     else if (((ts/1000) < lowerLimit) || ((ts/1000) > upperLimit)) {
+      _log.error(s"Rejecting row as timestamp: ${row.getLong(0)} is not within limits" )
```
Comment (vishramachandran) [inline core/src/main/scala/filodb.core/memstore/TimeSeriesPartition.scala]: Dont log. Just increase a kamon counter. Too many log messages will affect performance.

**Comment by velvia** [comment]
1) We don't need a lower limit.  There are legitimate use cases for lower time stamps, and a lower limit does not prevent ns/microsecond timestamps, which will always be higher.  
2) The check needs to be super super fast.  This is one of the most performance critical paths.  Absolutely cannot compute the time limit every single time.

**Pull request closed without merging**